sycl/include/sycl/detail/common.hpp

```diff
@@ -8,34 +8,32 @@
 
 #pragma once
 
+#include <array>                              // for array
+#include <cassert>                            // for assert
+#include <cstddef>                            // for size_t
+#include <string>                             // for allocator, operator+
 #include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
 #include <sycl/detail/export.hpp>             // for __SYCL_EXPORT
 #include <sycl/detail/pi.h>                   // for pi_int32
-
-#include <array>       // for array
-#include <cassert>     // for assert
-#include <cstddef>     // for size_t
-#include <string>      // for allocator, operator+
-#include <type_traits> // for enable_if_t
-#include <utility>     // for index_sequence, make_i...
+#include <sycl/detail/stl_type_traits.hpp>    // for is_output_iteratior
+#include <type_traits>                        // for enable_if_t
+#include <utility>                            // for index_sequence, make_i...
 
 // Default signature enables the passing of user code location information to
 // public methods as a default argument.
 namespace sycl {
 inline namespace _V1 {
 namespace detail {
 
-// The check for output iterator is commented out as it blocks set_final_data
-// with void * argument to be used.
-// TODO: Align these checks with the SYCL specification when the behaviour
-// with void * is clarified.
```

Comment:
Can you find where `void*` behavior was clarified? I would imagine we simply shouldn't allow it.

Reply:
Tagging some spec folks here: @gmlueck @Pennycook 
Do we allow set_final_data to be called with void * arguments?

Reply:
I might have done something wrong here, but it seems like `void*` doesn't satisfy `std::output_iterator`: https://godbolt.org/z/bhPM6ed4e.  I believe that in order for something to satisfy the iterator concept(s) it must be possible to dereference and increment it.  You can't dereference or increment a `void*`.

If my implementation and understanding are correct, then I don't think `set_final_data` should accept `void*`.

Reply:
Good to know, this is a surprise for me!
So now I see two options; either leave the full specialization of is_output_iterator with value = false **or** just delete this full specialization entirely and let the partial specialization do its thing, which is to emit an error whenever T = void *, in particular, cannot have "reference to void error". Id prefer the first because it makes it clear that void * is an exception to the rule.
Any feedback is welcomed!

Reply:
I think less is better. "void *" isn't any different from any other type that doesn't satisfy output iterator requirements.